### ISSUE TITLE
Allow to customize commit message

### ISSUE THREAD

**Issue by Liu233w** [dabutvin/Imgbot#619]
Hi,
It will be great if it's allowed to customize the commit message of imgbot, so that we can use [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/) without failing commitlint check.

**Comment by dabutvin** [comment]
Hi @Liu233w 
Love this idea! Are you thinking just to be able to customize the title of the commit message or the whole body as well? Do you see this being configured in the `.imgbotconfig` file?

**Comment by Liu233w** [comment]
Hi @dabutvin 
In my point of view, a customizable title is ok, because commitlint only check the title.

**Comment by daveabrock** [comment]
@dabutvin @Liu233w I could take a stab at this, at the ability to configure the title.

**Comment by dabutvin** [comment]
@daveabrock awesome! Let me know if you need any help getting started here or anything

**Comment by dabutvin** [comment]
also, check out #232 when you get in there for more background on the requests in this area

**Comment by daveabrock** [comment]
Hey @dabutvin - I have a PR almost ready, with it checking for a value from `.imgbotconfig` and populating the title in the commit message and the pull requests (with tests, of course). Any interest in commit formatting logic, like what is referenced in #232, or should we push out what we have and build off it as needed?

**Comment by dabutvin** [comment]
We don't need to solve everything all at once, if you got something we can ship then let's do it!

**Comment by daveabrock** [comment]
Any advice on accessing the `.imgbotconfig` file from `OpenPr.cs` to check for the commit title (or just passing down the title)? In `CompressImages.cs`, you grab the file like this before deserializing:

```
var repoConfigJson = File.ReadAllText(parameters.LocalPath + Path.DirectorySeparatorChar + ".imgbotconfig");
```

From what I understand, the `LocalPath` property's `parameters` object has come from a cloned temporary directory, but in the PR level, we don't have access to that. All my ideas are suuuuuper-hacky, any advice?

In `OpenAsync` in the `PullRequest` class, I take in a `RepoConfiguration` object but not sure how to pass it down from the top function level, if that makes sense.

**Comment by dabutvin** [comment]
sure!

Check out [Stats.cs](https://github.com/dabutvin/Imgbot/blob/master/OpenPrFunction/Stats.cs)
The commit message will have the commit title at this point and we can parse the PR title from the first line in the message.

Will probably have to return a different type so we can hold the title and all the stats

**Comment by daveabrock** [comment]
Ah! Got it now! Thanks.

**Comment by daveabrock** [comment]
I should probably run the function before submitting the PR and not relying on unit tests alone. :)

So do I need to generate these secrets in my own repo to test? The `APP_PRIVATE_KEY`, `PGP_PRIVATE_KEY`, `PGP_PASSWORD`, `TMP`?

**Comment by dabutvin** [comment]
Yep. I use a GitHub app just for local testing a lot. You can create one in the GitHub settings to get the app private key.

The pgp is just for commit signing and you can turn that off if you want.

Tmp is where the repo will get cloned

Let me know if you run into issues

**Comment by daveabrock** [comment]
Sorry @dabutvin, for all the newb questions but I've never registered a GitHub app before - any thoughts on how to "link" this to my repo? I see callback and hook URLs - how do you have it set up?

**Comment by dabutvin** [comment]
go to your apps landing page and configure it to be in a repo
for example, here is one i use with localhost a lot https://github.com/apps/imgbot-local

**Comment by moul** [comment]
Up

**Comment by daveabrock** [comment]
Yes! I hope to return to this one next week or two - it got away from me. Just need to work on testing it locally and whatnot, I think the code is good.

**Comment by moul** [comment]
Up :)

**Comment by Anmol368** [comment]
In my point of view, a customizable title is ok, because commitlint only check the title.

**Comment by daveabrock** [comment]
Sheesh, it's been 7 months hasn't it? I have a lot of time off this month and hope to code this soon. :)

**Comment by moul** [comment]
Not 100% sure if it's a good idea,  but having a custom body would allow auto DCO signing (and other git footer stuff)
